logagg_collector/collector.py

```diff
@@ -0,0 +1,926 @@
+
+        self._ensure_trackfiles_sync()
+
+
+    def register_to_master(self):
+        '''
+        Request authentication with master details
+        
+        Sample url: http://localhost:1088/logagg/v1/register_component?namespace=master
+                    &topic_name=logagg&host=localhost&port=1088
+        '''
+        #TODO: test case
+        master = self.master
+        url = self.REGISTER_URL.format(master_address=master.host+':'+master.port,
+                namespace=self.NAMESPACE,
+                topic_name=master.topic_name,
+                host=self.host,
+                port=self.port)
+
+        try:
+            register = requests.get(url)
+            register_result = json.loads(register.content.decode('utf-8'))
+            if register_result['success']:
+                if register_result['result']['success']:
+                    return register_result
+                else:
+                    err_msg = register_result['result']
+                    raise Exception(err_msg)
+            else:
+                err_msg = register_result['message']
+                raise Exception(err_msg)
+
+        except requests.exceptions.ConnectionError:
+            err_msg = 'Could not reach master, url: {}'.format(url)
+            raise Exception(err_msg)
+
+    def _init_fpaths(self):
+        '''
+        Files to be collected by default
+
+        >>> import tempfile
+        >>> temp_dir = tempfile.TemporaryDirectory()
+        >>> temp_dir_path = temp_dir.name
+        >>> trackfile = open(temp_dir_path+'/trackfiles.txt', 'w+'); trackfile.close()
+        >>> lc = collector = LogCollector('localhost', '1088', None, temp_dir_path, temp_dir_path)
+
+        >>> lc._init_fpaths()
+        [{'fpath': '/var/log/serverstats/serverstats.log', 'formatter': 'logagg_collector.formatters.docker_file_log_driver'}]
+
+        >>> temp_dir.cleanup()
+        '''
+        self.state['fpaths'] = [{'fpath':self.SERVERSTATS_FPATH,
+            'formatter':self.DOCKER_FORMATTER}]
+        self.state.flush()
+        return self.state['fpaths']
+
+
+    def _init_nsq_sender(self):
+        '''
+        Initialize nsq_sender on startup
+        >>> import tempfile
+        >>> temp_dir = tempfile.TemporaryDirectory()
+        >>> temp_dir_path = temp_dir.name
+        >>> trackfile = open(temp_dir_path+'/trackfiles.txt', 'w+'); trackfile.close()
+        >>> lc = collector = LogCollector('localhost', '1088', None, temp_dir_path, temp_dir_path)
+
+        >>> lc._init_nsq_sender() # doctest: +ELLIPSIS
+        (<deeputil.misc.Dummy object at ...>, <deeputil.misc.Dummy object at ...>)
+
+        >>> temp_dir.cleanup()
+        '''
+        #TODO: test cases for master mode
+
+        # Check if running on master mode or not
+        if self.master is None:
+            self.log.warn('nsq_not_set', msg='will send formatted logs to stdout')
+            return utils.DUMMY, utils.DUMMY
```

Comment:
If there is a return here you do not need to write else, you can directly continue writing code with the same indent space as if.